core/src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -66,6 +69,14 @@
     private final Engine.EventListener eventListener;
     private final QueryCache queryCache;
     private final QueryCachingPolicy queryCachingPolicy;
+    /**
+     * List of listeners for the engine to be created.
+     */
+    private final List<EngineCreationListener> creationListeners = new ArrayList<>();
+    /**
+     * List of listeners for the searcher being refreshed.
+     */
+    private final List<ReferenceManager.RefreshListener> refreshListeners = new ArrayList<>();
```

Comment:
can we please have only one listener and it should be final too. ie. pass it as a ctor argument?